webbpsf/tests/test_opds.py

```diff
@@ -22,3 +29,49 @@ def test_enable_adjustable_ote():
 
     assert ote.rms() > rms*10, "Huge piston offset didn't make the WFE much worse"
 
+
+
+GLOBAL_FOCUS = [-0.018251043541410904e-08]
+GLOBAL_FOCUS_SCALED = [-0.018251043541410904e-08 * 0.5]
+COEFFS_A1 = np.array([-3.52633363e-09, -2.90050902e-09, 1.25432196e-09, -7.43319098e-12,
+                      -5.82462948e-11, -1.27115922e-10, -1.91541104e-12, 3.64760396e-11,
+                      4.97176630e-13])
+thermal_model_parameters = ([1 * u.day, 'SM', GLOBAL_FOCUS, None],
+                            [1 * u.day, 'SM', GLOBAL_FOCUS_SCALED, .5],
+                            [1440, 'SM', GLOBAL_FOCUS, None],
+                            [1 * u.day, 'A1', COEFFS_A1, None],
+                            [0.0 * u.day, 'SM', [0.0], None],
+                            [0.0 * u.day, 'A1', np.zeros(9), None],
+                            [1.0 * u.day, 'D1', [0.0], None])
+@pytest.mark.parametrize('time, seg, coeff_truth, scaling', thermal_model_parameters)
+def test_thermal_slew_opd(time, seg, coeff_truth, scaling):
+    """ Test that the OTE Thermal model is outputting the correct values """
+    delta_time = time
+    # Create the thermal model
+    otelm = webbpsf.opds.OTE_Linear_Model_WSS()
+    otelm.delta_time = delta_time
+    otelm.start_angle = -5.
+    otelm.end_angle = 45.
+    otelm.scaling = scaling
+    otelm.thermal_model = webbpsf.opds.OteThermalModel(otelm.delta_time)
+    coeffs = otelm.thermal_slew_opd(segid=seg)
+    # Pull out coefficients
+    if isinstance(coeffs, float):
+        coeffs = [coeffs]
+    # Assert the coefficents
+        for coeff, truth in zip(coeffs, coeff_truth):
+            assert np.round(coeff, decimals=4) == np.round(truth, decimals=4)
+
```

Comment:
It would be nice to also add a test case that demonstrates the start_angle and end_angle parameter have the right effect if they're changed - it only seems they are used as their default values here.

Reply:
Agreed, I will add one